src/ZendeskApi.Client/Resources/UserIdentityResource.cs

```diff
@@ -28,9 +29,7 @@ public IResponse<UserIdentity> Post(UserIdentityRequest request)
 
         public IResponse<UserIdentity> Put(UserIdentityRequest request)
         {
-            var response = Post(request);
```

Comment:
To update a user's email address, we do the following:

I'm actually using that call to set/update the email. Here's the code
```
var userIdentities = _zendeskClient.UserIdentities.GetAll(user.Id.Value);
var emailIdentity = userIdentities.Results.FirstOrDefault(ui => ui.Type == "email");

if (emailIdentity == null)
{
    // User doesn't have an email set, so create one.
    _zendeskClient.UserIdentities.Post(new UserIdentityRequest()
    {
        Item = new UserIdentity()
        {
            UserId = user.Id,
            Type = "email",
            Value = email
        }
    });
}
else
{
    // User has an email set, so create update the value.
    emailIdentity.Value = email;
    _zendeskClient.UserIdentities.Put(new UserIdentityRequest()
    {
        Item = emailIdentity
    });
}
```

This requires the changes in this PR.